### PR TITLE
Two colors for tags

### PR DESCRIPTION
**Pull request by zadjii-msft**
Update the idl to allow us to specify separate FG and BG colors. I think that's better honestly.

### CODE REVIEW

File: src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowWalker/Components/ResultHelper.cs [hunk 1]
```diff
@@ -27,10 +26,10 @@ internal static List<WindowWalkerListItem> GetResultList(List<SearchResult> sear
     {
         if (searchControllerResults == null || searchControllerResults.Count == 0)
         {
-            return new List<WindowWalkerListItem>();
+            return [];
```
Comment (zadjii-msft) [inline src/modules/cmdpal/Exts/Microsoft.CmdPal.Ext.WindowWalker/Components/ResultHelper.cs:29]: gah most of this is just the code cleanup on save

File: src/modules/cmdpal/WindowsCommandPalette/TagViewModel.cs [hunk 2]
```diff
@@ -38,11 +52,11 @@ public Windows.UI.Color Color
 
     // TODO! VV These guys should have proper theme-aware lookups for default values
     // All this code is exceptionally terrible, but it's just here to keep the POC app running at this point.
-    internal Brush BorderBrush => new SolidColorBrush(Color);
+    internal Brush BorderBrush => new SolidColorBrush(Foreground);
 
-    internal Brush TextBrush => new SolidColorBrush(Color.A == 0 ? Windows.UI.Color.FromArgb(255, 255, 255, 255) : Color);
+    internal Brush TextBrush => new SolidColorBrush(Foreground.A == 0 ? Windows.UI.Color.FromArgb(255, 255, 255, 255) : Foreground);
```
Comment (zadjii-msft) [inline src/modules/cmdpal/WindowsCommandPalette/TagViewModel.cs:57]: i have not tested the POC app with this change, but c'est la vie

File: src/modules/cmdpal/doc/initial-sdk-spec/initial-sdk-spec.md [hunk 3]
```diff
@@ -1275,7 +1275,8 @@ block, and the generator will pull this into the file first.   -->
 interface ITag {
     IconDataType Icon { get; };
     String Text { get; };
-    OptionalColor Color { get; };
+    OptionalColor Foreground { get; };
```
Comment (michael-hawker) [inline src/modules/cmdpal/doc/initial-sdk-spec/initial-sdk-spec.md:1278]: Oh, does this generate the Microsoft.CmdPal.Extensions.idl file???

File: src/modules/cmdpal/WindowsCommandPalette/TagViewModel.cs (hunk 2 above)
Comment (michael-hawker) [reply src/modules/cmdpal/WindowsCommandPalette/TagViewModel.cs:57]: We could just leave this as a reference to the Background for the original behavior as we think we're close to switching anyway?

**Review by michael-hawker — APPROVED** [review]

**Comment by zadjii-msft** [comment]
it's usually best to commit code at 5:30pm on a friday

**Pull request merged**